graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java

```diff
@@ -2475,9 +2500,12 @@ private ForVariableDeclarationListResult variableDeclarationList(TokenType varTy
             final int varLine = line;
             final long varToken = Token.recast(token, varType);
 
-            // Get name of var.
-            final Expression binding = bindingIdentifierOrPattern(yield, await, CONTEXT_VARIABLE_NAME);
-            final boolean isDestructuring = !(binding instanceof IdentNode);
+            // Get left hand side.
+            // todo-lw: conditionalExpression feels way too broad here, but binding also uses it so idk
```

Comment:
`conditionalExpression` is indeed too broad here. not sure what you mean with "binding also uses it"?